client/src/app/site/agenda/components/agenda-sort/agenda-sort.component.ts

```diff
@@ -55,6 +69,16 @@ export class AgendaSortComponent extends BaseViewComponent {
         this.agendaRepo.sortItems(data).then(null, this.raiseError);
     }
 
+    /**
+     * Function to save the tree by click.
+     */
+    public async onSave(): Promise<void> {
+        const nodes = this._osSortTree.getTreeData();
```

Comment:
use `OSTreeNodeWithoutItem[]` as return type. Construct the OSTreeSortEvent as `{nodes: nodes}` (omit `parent_id`). This can be send to the server in a single request.